File: two-pointers/two-sum.js
const nums = [1,3,4,6,8,10,13];
const target = 6;

function twoSum(nums, target) {
	let left = 0;
	let right = nums.length - 1;

	while (left < right) {
		const sum = nums[right] + nums[left];

		if (sum === target) {
			return true;
		} else if (sum > target) {
			right--;	
		} else {
			left++;
		}
	}

	return false;
}

const res = twoSum(nums, target);

console.log(`nums: ${nums}; target: ${target}`);
console.log('Expected false:', res);

File: two-pointers/two-sum.py
def two_sum(nums, target):
    left = 0
    right = len(nums) - 1

    while left < right:
        sum = nums[left] + nums[right]
        print(f" left: {left}; right: {right}; sum: {sum};")

        if sum == target:
            return True

        if sum < target:
            left += 1
        else:
            right -= 1

    return False


nums = [1, 3, 4, 6, 8, 10, 13]
target = 6

res = two_sum(nums, target)
print(res)
